pkg/reconciler/mtbroker/trigger/trigger.go

```diff
@@ -163,7 +168,7 @@ func (r *Reconciler) subscribeToBrokerChannel(ctx context.Context, b *eventingv1
 		Name:       b.Name,
 		Namespace:  b.Namespace,
 	}
-	expected := resources.NewSubscription(t, brokerTrigger, brokerObjRef, uri, b.Spec.Delivery)
+	expected := resources.NewSubscription(t, brokerTrigger, brokerObjRef, uri, r.internalDeliveryConfigStore.Load())
```

Comment:
I'm little confused why we need to add the ConfigMap here. As discussed in the spec PR, I thought the logic here would simply be (something like this):
del := t.Spec.Delivery
if del == nil {
  del = b.Spec.Delivery
}
expected := resources.NewSubscription(t, brokerTrigger, brokerObjRef, uri, del)

if (t.Spec.Delivery != nil

Reply:
This subscription is an internal subscription between broker components, the actual dispatch to the service is done by the filter handler .

Now, with #4654 we're saying `BrokerSpec.Delivery` (same for `TriggerSpec.Delivery`) configures the trigger delivery, hence the hop between the filter_handler and the actual service. On the other hand, this subscription is an internal hop between broker components.

Reply:
Look at the description of this issue: https://github.com/knative/eventing/issues/4515